### PR TITLE
#203 - Refactor custom chapter using new principles

### LINKED ISSUE #203: Refactor custom chapter using new principles
## Background
Current unit tests are isolated methods.

## Feature
Migrate test structure from isolated methods to grouping test classes if needed.
Follow newly defined principles for python unit tests.
Apply as pilot of custom chapters only.

### PR DESCRIPTION
**Pull request by miroslavpojer**
## Summary by CodeRabbit

* **Documentation**
  * Added a new Python Unit Test Conventions principle and standardized function/class template wording.

* **Tests**
  * Reworked unit tests to a typed, fixture-driven approach and significantly expanded coverage for label handling, YAML parsing, edge cases, logging, and related behaviors.

* **Chores**
  * Updated project ignore rules to exclude common OS/editor and desktop artifacts.

Fixes #203

### CODE REVIEW

File: tests/unit/release_notes_generator/chapters/test_custom_chapters.py [hunk 1]
```diff
@@ -15,351 +15,321 @@
...
+    elif scenario == "empty_labels":
+        rec = record_stub("org/repo#Z", [])
+        records = {"org/repo#Z": rec}
+    else:
+        pytest.fail(f"Unhandled scenario: {scenario}")
+    # Act
     cc.populate(records)
-    assert not cc.chapters["Bugs"].rows, "Record without labels should be skipped"
+    # Assert
+    assert not cc.chapters["Bugs"].rows
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply tests/unit/release_notes_generator/chapters/test_custom_chapters.py]: Used.
